otcextensions/tests/functional/sdk/nat/v2/test_dnat.py

```diff
@@ -10,38 +10,169 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 import openstack
+import uuid
+
 from otcextensions.tests.functional import base
-from datetime import datetime
 
 _logger = openstack._log.setup_logging('openstack')
 
 
 class TestDnat(base.BaseFunctionalTest):
-    CURR_TIME = datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
-
-    def setUp(self):
-        super(TestDnat, self).setUp()
-
-    def test_list(self):
-        gateways = list(self.conn.nat.dnat_rules())
-        self.assertGreaterEqual(len(gateways), 0)
-
-    def test_list_filters(self):
-        attrs = {
-            'limit': 1,
-            'id': '2',
-            'project_id': '3',
-            'port_id': '4',
-            'private_ip': '5',
-            'internal_service_port': '6',
-            'floating_ip_id': '7',
-            'floating_ip_address': '8',
-            'external_service_port': '9',
-            'nat_gateway_id': '11',
-            'protocol': 'tcp',
-            'status': 'active',
-            'admin_state_up': True,
-            'created_at': self.CURR_TIME
-        }
-        gateways = list(self.conn.nat.dnat_rules(**attrs))
-        self.assertGreaterEqual(len(gateways), 0)
+
+    floating_ip = None
+    dnat_rule = None
+    port = None
+    uuid_v4 = uuid.uuid4().hex[:8]
+    port_name = uuid_v4 + 'test-dnat-port'
+    admin_external_net_id = "0a2228f2-7f8a-45f1-8e09-9039e1d09975"
```

Comment:
do not store id, for this you can use self.conn.get_network('admin_external_net')